pkg/authz/auth.go

```diff
@@ -1,149 +0,0 @@
-	Static                 []StaticAuthorizationConfig  `json:"static,omitempty"`
-}
-
-// SubjectAccessReviewRewrites describes how SubjectAccessReview may be
-// rewritten on a given request.
-type SubjectAccessReviewRewrites struct {
-	ByQueryParameter *QueryParameterRewriteConfig `json:"byQueryParameter,omitempty"`
-	ByHTTPHeader     *HTTPHeaderRewriteConfig     `json:"byHttpHeader,omitempty"`
-}
-
-// QueryParameterRewriteConfig describes which HTTP URL query parameter is to
-// be used to rewrite a SubjectAccessReview on a given request.
-type QueryParameterRewriteConfig struct {
-	Name string `json:"name,omitempty"`
-}
-
-// HTTPHeaderRewriteConfig describes which HTTP header is to
-// be used to rewrite a SubjectAccessReview on a given request.
-type HTTPHeaderRewriteConfig struct {
-	Name string `json:"name,omitempty"`
-}
-
-// ResourceAttributes describes attributes available for resource request authorization
-type ResourceAttributes struct {
-	Namespace   string `json:"namespace,omitempty"`
-	APIGroup    string `json:"apiGroup,omitempty"`
-	APIVersion  string `json:"apiVersion,omitempty"`
-	Resource    string `json:"resource,omitempty"`
-	Subresource string `json:"subresource,omitempty"`
-	Name        string `json:"name,omitempty"`
-}
-
-// StaticAuthorizationConfig describes what is needed to specify a static
-// authorization.
-type StaticAuthorizationConfig struct {
-	User            UserConfig
-	Verb            string `json:"verb,omitempty"`
-	Namespace       string `json:"namespace,omitempty"`
-	APIGroup        string `json:"apiGroup,omitempty"`
-	Resource        string `json:"resource,omitempty"`
-	Subresource     string `json:"subresource,omitempty"`
-	Name            string `json:"name,omitempty"`
-	ResourceRequest bool   `json:"resourceRequest,omitempty"`
-	Path            string `json:"path,omitempty"`
-}
-
-type UserConfig struct {
-	Name   string   `json:"name,omitempty"`
-	Groups []string `json:"groups,omitempty"`
-}
-
-// NewSarAuthorizer creates an authorizer compatible with the kubelet's needs
-func NewSarAuthorizer(client authorizationclient.SubjectAccessReviewInterface) (authorizer.Authorizer, error) {
-	if client == nil {
-		return nil, errors.New("no client provided, cannot use webhook authorization")
-	}
-	authorizerConfig := authorizerfactory.DelegatingAuthorizerConfig{
-		SubjectAccessReviewClient: client,
-		AllowCacheTTL:             5 * time.Minute,
-		DenyCacheTTL:              30 * time.Second,
-	}
-	return authorizerConfig.New()
-}
-
-type staticAuthorizer struct {
-	config []StaticAuthorizationConfig
-}
-
-func (saConfig StaticAuthorizationConfig) Equal(a authorizer.Attributes) bool {
-	isAllowed := func(staticConf string, requestVal string) bool {
-		if staticConf == "" {
-			return true
-		} else {
-			return staticConf == requestVal
-		}
-	}
-
-	userName := ""
-	if a.GetUser() != nil {
-		userName = a.GetUser().GetName()
-	}
-
-	if isAllowed(saConfig.User.Name, userName) &&
```

Comment:
Allowing an empty username (and really any field) in the file to match all users looks like a recipe for "mispelled a value and now any user can do this!"

TODO: post-acceptance, start a path to have a safe (non-matching) default and migrate with warnings, flags, and forced changes.


explicitly using a "*" instead is safer and doesn't match valid names.